src/vorta/borg/compact.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any, Dict
+from vorta.i18n import trans_late
+from vorta.utils import borg_compat
+from .borg_job import BorgJob
+
+
+class BorgCompactJob(BorgJob):
+
+    def started_event(self):
+        self.app.backup_started_event.emit()
+        self.app.backup_progress_event.emit(self.tr('Starting repository compaction...'))
+
+    def finished_event(self, result: Dict[str, Any]):
+        """
+        Process that the job terminated with the given results.
+
+        Parameters
+        ----------
+        result : Dict[str, Any]
+            The (json-like) dictionary containing the job results.
+        """
+        self.app.backup_finished_event.emit(result)
+        self.result.emit(result)
+        if result['returncode'] != 0:
+            self.app.backup_progress_event.emit(self.tr('Errors during compaction. See logs for details.'))
+        else:
+            self.app.backup_progress_event.emit(self.tr('Compaction completed.'))
+
+    @classmethod
+    def prepare(cls, profile):
+        ret = super().prepare(profile)
+        if not ret['ok']:
+            return ret
+        else:
+            ret['ok'] = False  # Set back to false, so we can do our own checks here.
+
+        if not borg_compat.check('COMPACT_SUBCOMMAND'):
+            ret['ok'] = False
+            ret['message'] = trans_late('messages', 'This feature needs Borg 1.2.0 or higher.')
+            return ret
+
+        cmd = ['borg', '--info', '--log-json', 'compact', '--cleanup-commits']
```

Comment:
`--cleanup-commits` was intended for one time use (per repo), afterwards i guess it is not really needed any more.

Reply:
We can keep it for a while until most repos are managed with 1.2 or cleaned up. This command won't run too often. Currently it's manually. Later we can run it together with prune every few weeks.

Reply:
Well, my point is that existence of 17byte files are not a bug in general. These are just separated commits.

In borg 1.1, there is a bug creating or leaving behind too many of them.
In borg 1.2 this bug was resolved, but separated commits are still created intentionally.

So I guess I rather would not remove them after the switch from 1.1 to 1.2.

Reply:
So this parameter option is only intended for very specific repositories that are known to be spammed with commits.

Reply:
it's intended to be run once (per repo) when borg is upgraded to 1.2.

Reply:
So would you remove it or keep it? What I’m probably asking: is there any harm in running it too often?

Reply:
Maybe vorta could detect whether a repository is still from 1.1 and run compact in this case.

Reply:
Well, that cleanup operation is rather simplistic: it just removes all separated commits (17b files). The compact_segments running afterwards will then add a commit at the end.

If the repo is ok before, it will also be ok afterwards, but IIRC I didn't analyze all error conditions that could also exist.

Reply:
besides that, if something goes wrong in a repo (like losing some files), it is nice to have multiple commits present.
if there is some commit before the lost files, one can just roll back to that commit and all is fine again.
you don't have that if you regularly remove all commits except the final one.

Reply:
Then I'll probably remove the `--cleanup-commits` option when touching the code next time. No need to overdo it.

> besides that, if something goes wrong in a repo (like losing some files), it is nice to have multiple commits present.

This is for `borg compact` in general? And it shouldn't run too often? I was planning to run it every few weeks, similar to our `borg check` option.

Reply:
matter of taste / requirements:
- more safety: run less often
- more space savings: run more often

Reply:
In that case I would add an additional option to the schedule tab like the one for validation but for compacting.